marker_api/src/sem/ty/fn_ty.rs

```diff
@@ -38,18 +39,20 @@ impl<'ast> FnTy<'ast> {
 /// closure. This type on it's own therefore only identifies the type of the closure.
 #[repr(C)]
 #[derive(Debug)]
+#[cfg_attr(feature = "driver-api", derive(typed_builder::TypedBuilder))]
 pub struct ClosureTy<'ast> {
-    closure_ty_id: TyDefId,
+    data: CommonTyData<'ast>,
+    ty_id: TyDefId,
     generics: GenericArgs<'ast>,
 }
 
 impl<'ast> ClosureTy<'ast> {
-    /// This returns the [`ItemId`] of the identified function.
+    /// This returns the [`ItemId`] of the struct that was generated for this closure.
     pub fn closure_ty_id(&self) -> TyDefId {
-        self.closure_ty_id
+        self.ty_id
```

Comment:
Consider renaming the method as well?

Reply:
The rename was mostly due to a clippy lint triggering. For ADT types, the `TyDefId` returned by `AdtTy::def_id()` is the id of the actual type, that it points to. Closures are similar, as they also point to a compiler generated type, which also has corresponding trait implementations. I'm not sure if it's better to name this `def_id()` which would indicate that this is similar to an ADT or something different, which better captures the kind of data this is and what it's used for :thinking:

Reply:
I can't think of any ambiguity as in what `ty_id` or `def_id` would return for the closure, unless you think there may be a method on the closure that may return a different type def ID

Reply:
Yeah, you're right. I now named it `def_id` to be inline with `AdtTy::def_id`.